packages/samples/headless-react/src/components/result-list/result-link.tsx

```diff
@@ -7,37 +7,22 @@ interface LinkProps {
 }
 
 export const ResultLink: FunctionComponent<LinkProps> = (props) => {
-  let wasOpened = false;
-  const onOpen = () => {
-    if (wasOpened) {
-      return;
-    }
-    wasOpened = true;
-    engine.dispatch(ResultAnalyticsActions.logDocumentOpen(props.result));
-  };
-
-  // 1 second is a reasonable amount of time to catch most longpress actions
-  const longpressDelay = 1000;
-  let longPressTimer: number;
-
-  const startPressTimer = () => {
-    longPressTimer = window.setTimeout(onOpen, longpressDelay);
-  };
-  const clearPressTimer = () => {
-    longPressTimer && clearTimeout(longPressTimer);
-  };
+  const resultSelectionHelpers = buildResultSelectionHelpers(
+    engine,
+    props.result
+  );
 
-  useEffect(() => clearPressTimer, []);
+  useEffect(() => () => resultSelectionHelpers.cancelPendingSelect(), []);
 
   return (
     <a
       href={props.result.clickUri}
-      onClick={onOpen}
-      onContextMenu={onOpen}
-      onMouseDown={onOpen}
-      onMouseUp={onOpen}
-      onTouchStart={startPressTimer}
-      onTouchEnd={clearPressTimer}
+      onClick={() => resultSelectionHelpers.select()}
+      onContextMenu={() => resultSelectionHelpers.select()}
+      onMouseDown={() => resultSelectionHelpers.select()}
+      onMouseUp={() => resultSelectionHelpers.select()}
+      onTouchStart={() => resultSelectionHelpers.beginDelayedSelect()}
+      onTouchEnd={() => resultSelectionHelpers.cancelPendingSelect()}
```

Comment:
Considering this component is now greatly simplified, could we remove it and have everything in just the result-list sample?

Reply:
Since we ideally want to call `cancelPendingSelect` when the component is removed (it's currently called in the useEffect of this component), I have a small preference for keeping it that way. Otherwise, both the class and functional `ResultList` components may have to keep track of the `ResultLinkAnalytics` controllers.

Reply:
I see, that's a good point